examples/js/fortunewhale.js

```diff
@@ -0,0 +1,168 @@
+module.exports = function () {
+    const generateFortune = fw.Task({
+        id: "generateFortune",
+        run: "fortune"
+    });
+
+    const whaleSays = fw.task({
+        run: "whalesay",
+        inputs: {
+            "body": "{ output('generateFortune') }"
+        },
+        requires: [
+            generateFortune
+        ]
+    });
+
+    return fw.Workflow({
+        output: generateFortune,
+        tasks: [
+            generateFortune,
+            whaleSays
+        ]
+    })
+};
+
+// Simple dependencies with task IDs in the requires
+module.exports = function () {
+    const generateFortune = fw.Task({
+        id: "generateFortune",
+        run: "fortune"
+    });
+
+    const whaleSays = fw.task({
+        run: "whalesay",
+        inputs: {
+            "body": "{ output('generateFortune') }"
+        },
+        requires: [
+            generateFortune.id
+        ]
+    });
+
+    return fw.Workflow({
+        output: generateFortune,
+        tasks: [
+            generateFortune,
+            whaleSays
+        ]
+    })
+};
+
+// Referencing a task in an expression without an ID.
+module.exports = function () {
+    const generateFortune = fw.Task({
+        run: "fortune"
+    });
+
+    const whaleSays = fw.task({
+        run: "whalesay",
+        inputs: {
+            "body": `{ output('${generateFortune.id}') }`,  // or:
+            "body2": output(generateFortune.id),            // or:
+            "body3": output(generateFortune),
+        },
+        requires: [
+            generateFortune.id
+        ]
+    });
+
+    return fw.Workflow({
+        output: generateFortune,
+        tasks: [
+            generateFortune,
+            whaleSays
+        ]
+    })
+};
+
+// Sequence
+module.exports = function () {
+    return fw.sequence([
+        fw.task({
+            id: "generateFortune", // optional ID
+            run: "fortune"
+        }),
+        fw.sleep(1000),
+        fw.task({
+            run: "whalesay",
+            inputs: {
+                "body": "{ output('generateFortune') }"
+            }
+        })
+    ])
+};
+
+// Expression using helper functions
+module.exports = function () {
+    return fw.sequence([
+        fw.task({
+            id: "generateFortune", // optional ID
+            run: "fortune"
+        }),
+        fw.sleep(1000),
+        fw.task({
+            run: "whalesay",
+            inputs: {
+                "body": "{ output('generateFortune') }"
+            }
+        })
+    ])
+};
+
+// Expression with a function
+module.exports = function () {
+    return fw.sequence([
+        fw.task({
+            id: "generateFortune", // <- ID is optional, but needed to reference the function
+            run: "fortune"
+        }),
+        fw.sleep(1000), // <- how to specify ID with these built-in functions?
+        fw.task({
+            run: "whalesay",
+            inputs: {
+                "body": function (scope) {
```

Comment:
Why is there a function here?  Would something like `fw.tasks["generateFortune"].output` work?

Reply:
This is an idea to avoid the need for the string-based javascript expressions. The function is evaluated like the expressions, without the, like you also pointed out, awkward javascript-string-in-javascript approach.

Reply:
`fw.tasks["generateFortune"].output` would work too. However it is a bit tricky because the javascript is parsed to the WorkflowSpec at build-time, meaning that you need somehow store the operation (of looking up the output), not actually getting the output (which would always be empty at buildtime of course).

Option 1: make the output return a function similar to this.
Option 2: make the output return the correct javascript-string-in-javascript
Option 3: store a special marker in the output field, which we can convert to the proper reference at buildtime

Reply:
Ok, makes sense at some level.  I think it might be confusing for the programmer to reason about what runs when -- which parts are build time and which parts are runtime.  The more I look at this the more I'm thinking JS async/await fits this use case better.